tests/e2e_logging.py

```diff
@@ -645,6 +645,12 @@ def test_historical_receipts(network, args):
             c.read().encode("ascii"), default_backend()
         )
 
+    cert_path = os.path.join(primary.common_dir, "networkcert.pem")
+    with open(cert_path, encoding="utf-8") as c:
+        network_cert = load_pem_x509_certificate(
+            c.read().encode("ascii"), default_backend()
+        )
+
     TXS_COUNT = 5
     network.txs.issue(network, number_txs=5)
     for idx in range(1, TXS_COUNT + 1):
```

Comment:
We've made significant changes around receipts/node certs since 1.x. I wonder if we could move the receipt verification to the more generic `logging_app.py` and verify the receipts of some (all?) transactions in the `lts_compatibility` test? For example, a 1.x ledger wouldn't have the node certificate in the receipt and the user would have to retrieve the certificate from elsewhere.

Reply:
I am very reluctant to do this because I think it would be unacceptably slow to run. I also don't like logging_app.py, I find it very opaque. I like to have the assertions inside the testcases directly, not offloaded to another module may or may not be checking what's right in the context of the test. That way lies flag madness in my experience.

Making the check work across 1.x and 2.x shouldn't be too difficult, it's just a matter of falling back to the old logic when the cert isn't in the receipt itself.

Reply:
As discussed, I will update the lts test so we do some cross-upgrade checks.